### ISSUE TITLE
Add IPIP testing for GoBGP

### ISSUE THREAD

**Issue by robbrockbank** [projectcalico/calicoctl#1581]
Currently only testing IPIP features for Bird - we need to add similar for IPIP.

This requires updating the GoBGP daemon (or possibly even Felix) to handle adding/removing an IP address for the IPIP tunnel.

**Comment by ishidawataru** [comment]
As mentioned in #1553, currently `allocate-ipip-addr` is not called for gobgp backend.
I think we have two options to do the identical stuff for gobgp backend.

1. move what `allocate-ipip-addr` is doing into `felix`. Both BGP backend can use the same  mechanism for IPIP addressing. `felix` is already managing the IPIP device. It would be natural to let it also manage its IP address.

2. make [this](https://github.com/projectcalico/calicoctl/blob/master/calico_node/allocateipip/allocate_ipip_addr.go) as a library and do something like below inside gobgp backend

```go
if cidrs := ipip.GetIPIPEnabledPoolCIDRs(c); len(cidrs) > 0 {
		ipip.EnsureHostTunnelAddress(c, nodename, cidrs)
	} else {
		ipip.RemoveHostTunnelAddr(c, nodename)
}
```

**Comment by robbrockbank** [comment]
Hi @ishidawataru,

I'm not sure about moving in to Felix at the moment as I think I'd like to keep IP allocation separate from Felix.

To implement this properly as a library function would require watchers on the main Client API which we don't have yet although that is in the works.

I've a suggestion for a temporary workaround that would suffice for now:
-  Use confd as we are doing for Bird BGP backend
-  In the rc.local (in calico/node) do the following if using GoBGP:
  -  Remove or rename all of the calico/confd/conf.d/*.toml files _except_ the tunl-ip.toml
  -  Start then confd service

This would invoke confd for managing the IPIP re-allocation.

This change would be isolated to calico/node which I quite like the idea of.

**Comment by ishidawataru** [comment]
@robbrockbank Sure, I can do that for a temporary workaround.
What do you think about the long-term solution?
Once IPIP support is finished, I'd like to work on putting `calico-bgp-daemon` into `felix` and make them a single binary. In this case, the workaround doesn't fit and I think we need to choose from two options I listed above anyway.

**Comment by robbrockbank** [comment]
@ishidawataru :  Combining binaries is something that we are considering doing but not by including any routing solution within the Felix code base.  We'd probably do it as a separate calico/node binary that imports in Felix and GoBGP etc.  All of that is very early stages at the moment though.

**Comment by ishidawataru** [comment]
@robbrockbank Ah, I got it. Thanks for the explanation.
I'll work on enabling confd with gobgp backend.

**Comment by robbrockbank** [comment]
Fixed by #1601
